Crucible.NPCs/Quest/CrucibleQuestRequirementConfig.cs

```diff
@@ -0,0 +1,49 @@
+// <copyright file="CrucibleQuestRequirementConfig.cs" company="RoboPhredDev">
+// This file is part of the Crucible Modding Framework.
+//
+// Crucible is free software; you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+// Crucible is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+// You should have received a copy of the GNU Lesser General Public License
+// along with Crucible; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+// </copyright>
+
+namespace RoboPhredDev.PotionCraft.Crucible.NPCs
+{
+    using RoboPhredDev.PotionCraft.Crucible.CruciblePackages;
+    using RoboPhredDev.PotionCraft.Crucible.GameAPI;
+    using RoboPhredDev.PotionCraft.Crucible.Yaml;
+
+    /// <summary>
+    /// Configuration specifying an addional quest requirement.
+    /// </summary>
+    [DuckTypeCandidate(typeof(CrucibleAdditionalEffectsQuestRequirementConfig))]
+    [DuckTypeCandidate(typeof(CrucibleMainIngredientQuestRequirementConfig))]
+    [DuckTypeCandidate(typeof(CrucibleMaxIngredientsQuestRequirementConfig))]
+    [DuckTypeCandidate(typeof(CrucibleNeedIngredientQuestRequirementConfig))]
+    [DuckTypeCandidate(typeof(CrucibleNoIngredientQuestRequirementConfig))]
+    [DuckTypeCandidate(typeof(CrucibleStrongPotionQuestRequirementConfig))]
+    [DuckTypeCandidate(typeof(CrucibleWeakPotionQuestRequirementConfig))]
+    public abstract class CrucibleQuestRequirementConfig : CruciblePackageConfigNode
```

Comment:
This got a bit absurd but it ended up working pretty well. If you check out https://github.com/AndrewFahlgren/potioncraft-crucible/wiki/Crucible-Package-Reference#quest-requirements you can see it got a little complicated to explain. I am not sure if there is a better way to do this which may be more strait forward to the modder to implement.

Reply:
The yaml api this creates looks pretty spot on to me, I think it's fine.